WPMedia/fusion-news-theme-blocks
# TMEDIA-194 - Ad Block reserve space

## Description

Add new custom field to ad-block which defaults to true - which ads logic to reserve space for the ad to fill to avoid layout shifting.

## Jira Ticket
- [TMEDIA-194](https://arcpublishing.atlassian.net/browse/TMEDIA-194)

## Acceptance Criteria
1. A new boolean Custom Field is added to the Ad Block called Reserve space for ad? 
    * If the value is true then the block should reserve enough space for the ad to fill in on client side load
    * If the value is false or not present then no change in behavior needs to happen, it behaves as it does now (not reserving space) 
    * The field should default to true
2. If there are multiple heights requested for the ad, it should reserve the smaller height. It should reserve the relevant height by breakpoint.

## Test Steps

1. Checkout this branch `git checkout TMEDIA-194-reserve-ad-space`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/ads-block`
3. Visit a page with ads on and verify page loads with space for the ad to fill, and then ads are loaded and do not cause layout shifting - http://localhost/pf/adstesting/?_website=the-gazette
4. Article page with Ads - http://localhost/2019/09/17/global-kitchen-sink-article/?_website=the-gazette
5. Edit the ads testing page to disabled reserve space and verify layout shifting is back - Not custom fields for existing blocks on a page do inherit default values.

## Effect Of Changes
### Before

Ads caused lot of layout shifting

<img width="762" alt="TMEDIA-194-before" src="https://user-images.githubusercontent.com/868127/115238380-87395a00-a115-11eb-9da3-a447f093377f.png">


### After

Reduced layout shifting - still see some layout shift based on what ad is being served in slot that have multiple sizes allocated to them

<img width="831" alt="TMEDIA-194-after" src="https://user-images.githubusercontent.com/868127/115238411-902a2b80-a115-11eb-9f54-e5d495ed6049.png">


## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
